### PR TITLE
Streaming XML parser and utilities.

### PR DESCRIPTION
**Pull request by drewhk**
Migrated from https://github.com/drewhk/akka-xml-stream

### CODE REVIEW

File: xmlparser/src/main/scala/com/drewhk/stream/xml/Xml.scala [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+   */
+  def coalesce(maximumTextLength: Int): Flow[ParseEvent, ParseEvent, NotUsed] =
+    Flow.fromGraph(new Coalesce(maximumTextLength))
+
+  /**
+   * A Flow that transforms a stream of XML ParseEvents. This stage filters out any event not corresponding to
+   * a certain path in the XML document. Any event that is under the specified path (including subpaths) is passed
+   * through.
+   */
+  def subslice(path: immutable.Seq[String]): Flow[ParseEvent, ParseEvent, NotUsed] =
```
Comment (ktoso) [inline xmlparser/src/main/scala/com/drewhk/stream/xml/Xml.scala]: it seems as if adding JavaDSL would actually be very simple?
just `getSubslice(List)`, `getCoalesce()`, `getParser` - done?

Comment (johanandren) [inline xmlparser/src/main/scala/com/drewhk/stream/xml/Xml.scala]: Java API for this one?

**Comment by johanandren** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM

**Pull request merged**